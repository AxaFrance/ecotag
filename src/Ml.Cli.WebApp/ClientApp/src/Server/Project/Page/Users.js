import React from 'react';
import Table from '@axa-fr/react-toolkit-table';
import EmptyArrayManager from "../../../EmptyArrayManager";

const UserLine = ({users, annotationUser}) =>{
    const user = users.find(user => user.nameIdentifier.toLowerCase() === annotationUser.nameIdentifier.toLowerCase())
    if(!annotationUser || !user){
        return null;
    }
   
   return (<Table.Tr key={user.id}>
        <Table.Td>{user.email}</Table.Td>
        <Table.Td>{annotationUser.numberAnnotations}</Table.Td>
    </Table.Tr>);
}

export const UserAnnotationsStatus = ({ users = [], numberAnnotationsByUsers=[] }) => {
  return (
    <div>
      <h2>Annotations</h2>
        <EmptyArrayManager items={numberAnnotationsByUsers} emptyArrayMessage="Il n'y a pas d'annotation sur ce projet pour l'instant.">
          <Table>
            <Table.Header>
              <Table.Tr>
                <Table.Th>Emails</Table.Th>
                <Table.Th>Nombre d&apos;annotations</Table.Th>
              </Table.Tr>
            </Table.Header>
            <Table.Body>
                
                    {numberAnnotationsByUsers.map(annotationUser => (
                        <UserLine users={users} annotationUser={annotationUser} />
                    ))}
                
            </Table.Body>
          </Table>
        </EmptyArrayManager>
    </div>
  );
};

export default UserAnnotationsStatus;
