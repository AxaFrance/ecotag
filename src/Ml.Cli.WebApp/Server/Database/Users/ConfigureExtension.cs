using Microsoft.EntityFrameworkCore;
using Microsoft.Extensions.Configuration;
using Microsoft.Extensions.DependencyInjection;

namespace Ml.Cli.WebApp.Server.Database.Users;

public static class ConfigureExtension
{
    public static void ConfigureUserRattachment(this IServiceCollection services, IConfiguration configuration)
    {
        services.AddDbContext<UserContext>(options =>
            options.UseSqlServer(configuration.GetConnectionString("UserContext")));
        services.AddScoped<UserContext, UserContext>();
        services.AddScoped<IUsersRepository, UsersRepository>();
    }
}