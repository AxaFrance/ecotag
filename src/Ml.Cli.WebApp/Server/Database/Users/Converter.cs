using Ml.Cli.WebApp.Server.Groups.Database;

namespace Ml.Cli.WebApp.Server.Database.Users;

public static class Converter
{
    public static UserDataModel ToUserDataModel(this UserModel userModel)
    {
        return new UserDataModel
        {
            Id = userModel.Id.ToString(),
            Email = userModel.Email
        };
    }
}