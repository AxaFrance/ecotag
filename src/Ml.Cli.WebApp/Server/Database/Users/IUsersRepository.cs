using System.Threading.Tasks;

namespace Ml.Cli.WebApp.Server.Database.Users;

public interface IUsersRepository
{
    Task<UserDataModel> GetUserAsync(string id);
    
    Task<UserDataModel> GetUserByEmailAsync(string email);
    
    Task<string> CreateUserAsync(string email);
}