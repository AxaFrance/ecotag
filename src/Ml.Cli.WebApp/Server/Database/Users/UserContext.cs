using System.Threading.Tasks;
using Microsoft.EntityFrameworkCore;
using Ml.Cli.WebApp.Server.Groups.Database;

namespace Ml.Cli.WebApp.Server.Database.Users;

public class UserContext : DbContext
{
    public UserContext() { }

    public UserContext(DbContextOptions<UserContext> options) : base(options)
    {
    }

    public UserContext(DbContextOptionsBuilder options) : base(options.Options)
    {
    }

    protected override void OnConfiguring(DbContextOptionsBuilder optionsBuilder)
    {
        if (!optionsBuilder.IsConfigured)
        {
            optionsBuilder.UseSqlServer(
                "Data Source=localhost,1433;Initial Catalog=EcotagContent;Integrated Security=False;User ID=sa;Password=Your_password123;MultipleActiveResultSets=True");
        }
    }
    
    public virtual DbSet<UserModel> Users { get; set; }

    public Task<int> SaveChangesAsync()
    {
        return SaveChangesAsync(default);
    }
}