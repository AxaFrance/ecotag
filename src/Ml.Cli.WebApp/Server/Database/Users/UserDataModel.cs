namespace Ml.Cli.WebApp.Server.Database.Users;

public class UserDataModel
{
    public string Id { get; set; }
    public string Email { get; set; }
}