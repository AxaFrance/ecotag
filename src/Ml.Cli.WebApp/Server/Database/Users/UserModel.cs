using System;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace Ml.Cli.WebApp.Server.Database.Users;

[Table("T_User", Schema = "sch_etg")]
public class UserModel
{
    [Key]
    [Column("USR_Id")]
    public Guid Id { get; set; }
    
    [Column("USR_Email")]
    public string Email { get; set; }
}