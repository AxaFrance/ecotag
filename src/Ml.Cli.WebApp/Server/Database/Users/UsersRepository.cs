using System;
using System.Threading.Tasks;
using Microsoft.EntityFrameworkCore;
using Ml.Cli.WebApp.Server.Groups.Database;

namespace Ml.Cli.WebApp.Server.Database.Users;

public class UsersRepository : IUsersRepository
{
    private readonly GroupContext _groupsContext;

    public UsersRepository(GroupContext groupsContext)
    {
        _groupsContext = groupsContext;
    }
    
    public async Task<UserDataModel> GetUserAsync(string id)
    {
        var user = await _groupsContext.Users.AsNoTracking().FirstOrDefaultAsync(u => u.Id == new Guid(id));
        return user?.ToUserDataModel();
    }
    
    public async Task<UserDataModel> GetUserByEmailAsync(string email)
    {
        var user = await _groupsContext.Users.AsNoTracking().FirstOrDefaultAsync(u => u.Email == email.ToLower());
        return user?.ToUserDataModel();
    }

    public async Task<string> CreateUserAsync(string email)
    {
        var userModel = new UserModel
        {
            Email = email.ToLower()
        };
        _groupsContext.Users.Add(userModel);
        await _groupsContext.SaveChangesAsync();
        return userModel.Id.ToString();
    }
}