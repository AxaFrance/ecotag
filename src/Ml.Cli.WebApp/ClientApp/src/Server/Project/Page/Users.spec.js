import '@testing-library/jest-dom/extend-expect';
import React from 'react';
import { render, waitFor } from '@testing-library/react';
import {User} from './Users';
import {BrowserRouter as Router} from "react-router-dom";

describe('Users', () => {
  it('Users render correctly', async () => {
    const { asFragment, getByText } = render(<Router><User users={[{'email':'titi'}, {'email':'toto'}]} /></Router>);
    const messageEl = await waitFor(() => getByText('Informations annotateurs'));
    expect(messageEl).toHaveTextContent(
        'Informations annotateurs'
    );
    expect(asFragment()).toMatchSnapshot();
  });
});

