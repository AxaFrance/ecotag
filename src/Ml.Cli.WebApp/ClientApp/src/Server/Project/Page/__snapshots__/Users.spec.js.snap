// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Users Users render correctly 1`] = `
<DocumentFragment>
  <div>
    <h2>
      Annotations
    </h2>
    <span
      class="empty-array__error"
    >
      <b>
        Il n'y a pas d'annotation sur ce projet pour l'instant.
      </b>
    </span>
    <table
      class="af-table"
    >
      <thead
        class="af-table-thead"
      >
        <tr
          class="af-table__tr"
        >
          <th
            class="af-table__th"
          >
            Emails
          </th>
          <th
            class="af-table__th"
          >
            Nombre d'annotations
          </th>
        </tr>
      </thead>
      <tbody
        class="af-table__body"
      />
    </table>
  </div>
</DocumentFragment>
`;
